rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -966,11 +942,11 @@ def _prepare_layers(self) -> None:
         else:
             self._transformer = lambda x, mask, training: x
 
-        self._embed = {}
-
-        self.train_metrics = {"t_loss": tf.keras.metrics.Mean(name="t_loss")}
-        self.eval_metrics = {"val_t_loss": tf.keras.metrics.Mean(name="val_t_loss")}
+        self._prepare_mask_lm_layers()
+        self._prepare_intent_classification_layers()
+        self._prepare_entity_recognition_layers()
```

Comment:
I think it makes sense to create another method for sequence related staff above.
Shall we use a `dict` for all the layers including losses? Then we don't need `else`. Either general, or separate for each different method, not sure which approach is better. `self._layers` is occupied though.

Reply:
and the metric that calculates f1 score for crf also can be inside this dict

Reply:
Are you sure that `self._layers` is occupied?

Reply:
oh, they might be occupied only for `tf.keras.layers.Layer`

Reply:
also used for `tf.keras.models.Model`